device/device.go

```diff
@@ -35,8 +40,27 @@ type Device struct {
 	m                mqtt.Client
 	interfaces       map[string]interfaces.AstarteInterface
 	astarteAPIClient *client.Client
-	// AutoReconnect sets whether the device should reconnect automatically
+	brokerURL        string
+	// AutoReconnect sets whether the device should reconnect automatically. Defaults to false.
 	AutoReconnect bool
+	// ConnectRetry sets whether the device should retry to connect if the first connection
+	// fails. Defaults to false.
+	ConnectRetry bool
+	// MaxRetries sets the number of attempts for the device to establish the first connection.
+	// If ConnectRetry is false, MaxRetries will be ignored. Defaults to 10.
+	MaxRetries int
+	// InitialInterval sets the initial delay before a new connection attempt is performed.
+	// If ConnectRetry is false, InitialInterval will be ignored. Defaults to 500 ms.
+	InitialInterval time.Duration
+	// RandomizationFactor defines the range in which the next backoff will be set, according to
+	// randomized interval = RetryInterval * (random value in range [1 - RandomizationFactor, 1 + RandomizationFactor]).
+	// Defaults to 0.5.
+	RandomizationFactor float64
+	// Multiplier set the multiplicative coefficient used to determine the next retry interval.
+	// Defaults to 1.5.
+	Multiplier float64
+	// MaxInterval caps the RetryInterval (not the randomized interval). Defaults to 60 s.
+	MaxInterval time.Duration
```

Comment:
Exposing this retry details to the end user can lead to users shooting themselves in the foot. I'd expose only `MaxRetries` leaving everything else fixed to the values that were chosen as default.